File: aspnet-core/src/Event_Planning_System.Core/Entities/Event.cs
using Abp.Domain.Entities;
using Abp.Domain.Entities.Auditing;
using Event_Planning_System.Authorization.Users;
using Event_Planning_System.Entities;
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
public enum EventCategory
{
    Concert,
    Conference,
    Workshop,
    Seminar,
    Party,
    Exam,
    Birthday,
    Graduation,
    Baby_Shower,
    Wedding,
    Gathering,
    Other
}

public enum EventType
{
    Paid,
    Free
}


namespace Event_Planning_System.Enitities
{
    public class Event : IEntity<int>
    {

        public int Id { get; set; }
        [Required(ErrorMessage = "Event name is required.")]
        [StringLength(100, ErrorMessage = "Event name can't be longer than 100 characters.")]
        public string Name { get; set; }

        [StringLength(500, ErrorMessage = "Description can't be longer than 500 characters.")]
        public string Description { get; set; }
        public bool isRead { get; set; } = false;
		[Required(ErrorMessage = "Location is required.")]
        [StringLength(200, ErrorMessage = "Location can't be longer than 200 characters.")]
        public string Location { get; set; }

        [Required(ErrorMessage = "Start date is required.")]
        [DataType(DataType.Date)]
        public DateTime StartDate { get; set; }

        [Required(ErrorMessage = "End date is required.")]
        [DataType(DataType.Date)]
        public DateTime EndDate { get; set; }

        public bool IsPublic { get; set; }

        
        private int _maxCount;

        [Range(1, 10000, ErrorMessage = "Max count must be between 1 and 10000.")]
        public int MaxCount
        {
            get => _maxCount;
            set
            {
                _maxCount = value;
                NumberOfTickets = value;
            }
        }

        public int NumberOfTickets { get; set; }

        [RegularExpression(@"^.+\.(png|jpg|jpeg)$", ErrorMessage = "Image must be in PNG, JPG, or JPEG format.")]
        public string EventImg { get; set; }

        public EventCategory Category { get; set; }

        public long UserId { get; set; }
        [ForeignKey("UserId")]
        public virtual User User { get; set; }

        public EventType Type { get; set; }

        public int? TicketPrice { get; set; }

        public virtual ICollection<notification> Notifications { get; set; } = new List<notification>();
        public virtual ICollection<Guest> Guests { get; set; } = new List<Guest>();

        public virtual ICollection<BudgetExpense> Budgets { get; set; } = new List<BudgetExpense>();

        public virtual ICollection<ToDoCheckList> ToDoCheckLists { get; set; } = new List<ToDoCheckList>();
        public virtual ICollection<Feedback> Feedbacks { get; set; } = new List<Feedback>();
		public virtual ICollection<GuestsFeedback> GuestsFeedback { get; set; } = new List<GuestsFeedback>();
		public ICollection<GuestEvent> GuestEvents { get; set; }
        public ICollection<FavoriteEvent> FavoriteEvents { get; set; } = new List<FavoriteEvent>();
        public int? PlaceId { get; set; }
        [ForeignKey("PlaceId")]
        public virtual SupplierPlaces SupplierPlaces { get; set; }
        public bool RequestPlace { get; set; }

        public bool IsTransient()
        {
            throw new NotImplementedException();
        }
    }
}

File: angular/src/shared/Models/Event.ts
export enum Enumerator {
    Concert = "Concert",
    Conference = "Conference",
    Workshop = "Workshop",
    Seminar = "Seminar",
    Party = "Party",
    Exam = "Exam",
    Birthday = "Birthday",
    Graduation = "Graduation",
    Baby_Shower = "Baby Shower",
    Wedding = "Wedding",
    Gathering = "Gathering",
    Other = "Other"
}
export enum EventType {
    Paid = 0,
    Free = 1
}
export enum PlaceState  {
    Pendding=0,
    Rejected=1,
    Accepted=2
}


export class Event {
    constructor(
        public id: number = 0,
        public isRead: boolean | undefined = false,
        public name: string | undefined = undefined,
        public description: string | undefined = undefined,
        public location: string | undefined = undefined,
        public startDate: Date | undefined = undefined,
        public endDate: Date | undefined = undefined,
        public isPublic: boolean | undefined = undefined,
        public maxCount: number | undefined = undefined,
        public eventImg: string | undefined = undefined,
        public eventImgFile: File | undefined | null = null,
        public category: string | undefined = undefined,
        public userId: number | undefined = undefined,
        public budgetId: number | undefined = undefined,
        public type: EventType = EventType.Free,
        public ticketPrice: number | undefined = undefined,
        public numberOfTickets :number |undefined =undefined,
        public source?: 'attending' | 'upcoming',
        public placeId: number | undefined = undefined,
        public requestPlace: PlaceState | undefined =0,
        public placeName : string | undefined = undefined,
        public userName : string |undefined=  undefined,
        public userEmail :string |undefined=  undefined,
        public ContactEmail :string |undefined=undefined,
    ) {}
}
